src/uct/ib/rc/accel/rc_mlx5_common.h

```diff
@@ -39,16 +39,20 @@
 
 #define UCT_RC_MLX5_ATOMIC_FOPS (UCT_RC_MLX5_ATOMIC_OPS | UCS_BIT(UCT_ATOMIC_OP_SWAP))
 
-#define UCT_RC_MLX5_CHECK_ATOMIC_OPS(_op)                               \
-    if (ucs_unlikely(!(UCS_BIT(_op) & UCT_RC_MLX5_ATOMIC_OPS))) {       \
-        ucs_assertv(0, "incorrect opcode for atomic_post: %d", opcode); \
-        return UCS_ERR_UNSUPPORTED;                                     \
+#define UCT_RC_MLX5_CHECK_ATOMIC_OPS(_op, _size)                                \
+    if (ucs_unlikely(!(UCS_BIT(_op) & UCT_RC_MLX5_ATOMIC_OPS))) {               \
+        ucs_assertv(0, "incorrect opcode for atomic_post: %d", _op);            \
+        return UCS_ERR_UNSUPPORTED;                                             \
+    } else {                                                                    \
+        ucs_assert((_size == sizeof(uint64_t)) || (_size == sizeof(uint32_t))); \
     }
 
-#define UCT_RC_MLX5_CHECK_ATOMIC_FOPS(_op)                              \
-    if (ucs_unlikely(!(UCS_BIT(_op) & UCT_RC_MLX5_ATOMIC_FOPS))) {      \
-        ucs_assertv(0, "incorrect opcode for atomic_post: %d", opcode); \
-        return UCS_ERR_UNSUPPORTED;                                     \
+#define UCT_RC_MLX5_CHECK_ATOMIC_FOPS(_op, _size)                               \
+    if (ucs_unlikely(!(UCS_BIT(_op) & UCT_RC_MLX5_ATOMIC_FOPS))) {              \
+        ucs_assertv(0, "incorrect opcode for atomic_post: %d", _op);            \
```

Comment:
btw, it can be united with UCT_RC_MLX5_CHECK_ATOMIC_OPS, if you use the 3d parameter. 
The code is the same.

Reply:
wording was bit-wrong, updated